app/controllers/locations_controller.rb

```diff
@@ -1,14 +1,33 @@
 class LocationsController < ApplicationController
-  def index
-    @locations = Location.all
-  end
+  before_action :authenticate_user, except: [:index, :show]
```

Comment:
You can remove `, except: [:index, :show]` since these actions don't exist here anymore